File: backend/app/models/reminder.py
from sqlalchemy import Column, BigInteger, String, Date, DateTime, Text, JSON
from sqlalchemy.sql import func

from app.core.database import Base


class Reminder(Base):
    __tablename__ = "t_reminder"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    reminder_id = Column(String(32), unique=True, nullable=False, index=True)
    user_id = Column(String(32), nullable=False, index=True)
    contact_id = Column(String(32), nullable=False)
    type = Column(String(20), nullable=False)
    anniversary_id = Column(String(32), nullable=True)
    holiday_id = Column(String(32), nullable=True)
    event_title = Column(String(50), nullable=False)
    event_date = Column(Date, nullable=False)
    remind_time = Column(DateTime, nullable=False)
    status = Column(String(20), default="unread")
    blessing = Column(Text, nullable=True)
    gifts = Column(JSON, nullable=True)
    created_at = Column(DateTime, server_default=func.now())

File: mobile/EternalMoments/Models/Reminder.swift
import Foundation

struct ReminderListData: Decodable {
    let total: Int
    let list: [Reminder]
}

struct Reminder: Codable, Identifiable, Hashable {
    let id = UUID()
    let reminderId: String
    let contactId: String
    let type: String
    let anniversaryId: String?
    let holidayId: String?
    let eventTitle: String
    let eventDate: String
    let remindTime: String
    let status: String
    let blessing: String?
    let gifts: [Gift]?

    enum CodingKeys: String, CodingKey {
        case reminderId = "reminder_id"
        case contactId = "contact_id"
        case type
        case anniversaryId = "anniversary_id"
        case holidayId = "holiday_id"
        case eventTitle = "event_title"
        case eventDate = "event_date"
        case remindTime = "remind_time"
        case status
        case blessing
        case gifts
    }
}

struct Gift: Codable, Hashable {
    let name: String
    let price: Double
    let reason: String
    let purchaseUrl: String

    enum CodingKeys: String, CodingKey {
        case name
        case price
        case reason
        case purchaseUrl = "purchase_url"
    }
}
